src/update.rs

```diff
@@ -0,0 +1,182 @@
+use std::convert::TryInto;
+use std::convert::TryFrom;
+
+use async_trait::async_trait;
+
+use google_datastore1::schemas::{
+  BeginTransactionRequest, BeginTransactionResponse, CommitRequest, CommitResponse, Entity, Key,
+  Mutation, MutationResult,
+};
+
+use crate::entity::{DatastoreEntity};
+
+use crate::error::{DatastorersError, DatastoreClientError};
+use crate::connection::DatastoreConnection;
+
+#[async_trait]
+pub trait DatastorersUpdatable<E> {
+
+  async fn commit(self, connection: &(impl DatastoreConnection + Send + Sync)) -> Result<E, DatastorersError>;
```

Comment:
I need Send + Sync on the connection here, I guess that is since it can be sent to an other threads, but we do not have that in any other place... I mainly wonder why we only need it here

Reply:
I think because of the async_trait macro. This limit will probably go away when async fn in traits are fixed in the language